src/circllhist.c

```diff
@@ -930,6 +940,38 @@ hist_remove(histogram_t *hist, double val, uint64_t count) {
   return 0;
 }
 
+uint64_t
+hist_remove_raw(histogram_t *hist, hist_bucket_t hb, uint64_t count) {
+  int idx;
+  ASSERT_GOOD_HIST(hist);
+  if(hist_internal_find(hist, hb, &idx)) {
+    uint64_t newval = hist->bvs[idx].count - count;
+    if(newval > hist->bvs[idx].count) newval = 0; /* we rolled */
```

Comment:
Nit: I find this after the fact checking for rolling confusing.
Is there a reason not to check do this up-front, like so: ?
```
if(hist->bvs[idx].count <= count) {
  uint64_t newval = hist->bvs[idx].count;
  hist->bvs[idx].count = 0;
  return newval;
}
else {
  hist->bvs[idx].count = hist->bvs[idx].count - count;
  return count;
}
```